docs/contracts.rst

```diff
@@ -496,9 +505,17 @@ Pure Functions
 
 Functions can be declared ``pure`` in which case they promise not to read from or modify the state.
 
+In addition to the list of state modifying statements explained above, the following are reading from the state:
+#. Reading from state variables.
+#. :ref:`Emitting events. <events>`
+#. Accessing ``this.balance`` or ``<address>.balance``
+#. Accessing any of the members of ``tx``, ``msg`` or ``block``.
```

Comment:
Apart from `msg.sender`, `msg.data` and `msg.value`. Actually all `msg` members should be fine, shouldn't they?

Reply:
Ah,gas is not.

Reply:
I guess only `msg.sig` is safe.